Bottom: d51ee3c05a8843bddf1be206a39787057795899e
Top:    75b81b1c973de862de24a94d944dc4d530462c21
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-23 19:41:24 +0000

Workaround crash where contour is passed as NULL to poly_PreContour








---

diff --git a/src/polygon.c b/src/polygon.c
index 85720c1..80f4cb4 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
